terraform/modules/securityhub/main.tf

```diff
@@ -1,22 +1,108 @@
+################
+# Security Hub #
+################
+
+# Get current region
 data "aws_region" "current" {}
 
-# Enable SecurityHub
-resource "aws_securityhub_account" "default" {}
+# Get the delegated administrator account ID
+data "aws_caller_identity" "delegated-administrator" {
+  provider = aws.delegated-administrator
+}
+
+####################################
+# Security Hub in the root account #
+####################################
+resource "aws_securityhub_account" "default" {
+  provider = aws.root-account
+}
 
 # Enable Standard: AWS Foundational Security Best Practices
-resource "aws_securityhub_standards_subscription" "aws-foundational" {
+resource "aws_securityhub_standards_subscription" "default-aws-foundational" {
   standards_arn = "arn:aws:securityhub:${data.aws_region.current.name}::standards/aws-foundational-security-best-practices/v/1.0.0"
```

Comment:
They've rarely changed in the past, and Terraform doesn't have a way to get these via a data source at the moment. Even when Security Hub adds new _controls_ within a standard, the version listed here aligns with the actual standard version, rather than the controls that are part of it.

For example, the [AWS Foundational Security Best Practices was updated two days ago](https://aws.amazon.com/about-aws/whats-new/2021/03/aws-security-hub-adds-25-new-controls-to-foundational-security-best-practices-standard/) with 25 new controls and doesn't change the actual standard version.